### ISSUE TITLE
Mac Sequoia: Download rate deteriorating to near zero after a few seconds at normal speed

### ISSUE THREAD

**Issue by Dondo11** [openaudible/openaudible#1460]
This happens on macOS Sequoia 15.0 and OpenAudible 4.4.6

Downloads commence at normal speed up until about 4% of the audiobook was been downloaded, then drops to an intermittent rate of a few kB/s, then halts altogether.

I think this also happened with 4.4.5, so may be related to Sequoia; other downloads are at normal speed, however.

**Comment by openaudible** [comment]
I just upgraded to Sequoia and am unable to see that problem. It was a steady 10MB/sec for an entire large book. 

If it halts altogether, perhaps you can get an error message from the File: Log Window?

You might try fast.com while downloading to see if there is anything else.

Usually there are no anti-virus apps, but if you have one, make sure you give OpenAudible network permissions.

If anyone else sees this, please post..

**Comment by Dondo11** [comment]
Thanks for your speedy response to this.

I've been trying a few things and noticed that one other app is showing similar behaviour, albeit with a much less drastical drop in download speed.

I downloaded the AAX file directly from Audible via Safari and this worked at full speed without a hitch.  Import in OpenAudible and coversion to m4b worked as expected.

Maybe related to the firewall issue observed by others? —> https://techissuestoday.com/macos-sequoia-messages-and-firewall-issues/

**Comment by openaudible** [comment]
Do you have a firewall? Can you disable it and try again? Sounds like you found the problem— Apple will probably fix.

**Comment by Dondo11** [comment]
This is interesting indeed!  And it's obviously a Mysterious Case of the Firewall 😋

I gave it another try again today since the other app I mentioned was behaving normally again.  No such luck, though, with OpenAudible:  Download went normal up until about 13%, then rapidly slowed down, stalled altogether.

Deactivated the macOS firewall.  (You can try playing with this yourself, by the way, and toggle it via Settings->Network->Firewall.)

Lo and behold:  Problem gone, OpenAudible download back to normal.  To make sure this wasn't just a glitch with this particular book, I took the following steps:

- turned the fire wall back on
- started downloading a different book, which stalled as above
- quit OpenAudible
- turned firewall off
- started OpenAudible
- downloaded the book whose download had failed
- Success!!

So we can safely assume it's a firewall-related issue which miraculously doesn't occur when using Safari, i.e. an Apple app.

You may want to try this yourself, then log a complaint with Apple.

Thanks for your time and apologies should I have caused any upset!

Please let me know if I should close this issue (or perhaps you can do this yourself?)!

**Comment by oboe45** [comment]
I use an Imac, use either Safari or Chrome.  Downloads on either browser
 with current OA version 4.4.6, are consistently fast, beginning to end
with no noticeable variance after a dozen or more repeats on same files
(some large, some small), individually and in groups, using Sonoma 14.6.1

**Comment by Dondo11** [comment]
That‘s basically my observation:  Browser downloads work just fine while quite a few apps seem to experience issues due to the macOS firewall being active.  Deactivate the firewall and things behave normal.

Enter „macOS Sequoia firewall“ in any search engine and find lots of hits for this general issue.  OpenAudible isn’t the culprit.

**Comment by openaudible** [comment]
Is there another 3rd party security product installed on this machine? Seems like most of the reports are for people using Crowdstrike or similar products. 

I tested with the default Sequoia firewall turned on with default settings and downloads worked. I did notice a constant 70% performance decrease with the firewall on. That seems bad, but I haven't tried with earlier versions of MacOS to see if that is normal. 

I also tried blocking all incoming connections and downloads continued to work for entire books. 

Will keep this open in case others experience similar issues.. but will assume Apple will fix this..

**Comment by Dondo11** [comment]
Yes, I'm using Liitle Snitch, too, which may seem like a bit of overkill when the macOS firewall is also active.  Indeed, my first suspect in this was Little Snitch, particularly because I'd updated to its latest version briefly after updating to Sequoia.  When the ebook download via OpenAudible stalled I first deactivated Little Snitch, but the behaviour didn't change at all.  Then I discovered the apparent macOS firewall issues others seem to experience and deactivted the macOS firewall whilst keeping Little Snitch active.  In this setting, everything works fine, so I'll keep Little Snitch and chuck macOS firewall for the time being.  Hopes are not high for Apple to fix whatever this is since the issue has apparently been around even before Sequoia.

I suspect there may be a more general networking issue in Sequoia because the IMAP inboxes of my e-mail accounts don't properly synchronise across my Apple devices ever since I updated to iOS 18/iPadOS 18/Sequoia.  For instance, a message deleted on one device will stay visible on another device, and IMAP server retrieval stays in "pending" state even after retrieving message headers.
